cloudinit/config/cc_grub_dpkg.py

```diff
@@ -24,7 +24,7 @@
     "id": "cc_grub_dpkg",
     "distros": ["ubuntu", "debian"],
     "frequency": PER_INSTANCE,
-    "activate_by_schema_keys": [],
+    "activate_by_schema_keys": ["grub_dpkg", "grub-dpkg"],
```

Comment:
We still need to resolve https://github.com/canonical/cloud-init/issues/5842, but I can do that separately.